test/Integration/SqlTriggerBindingIntegrationTests.cs

```diff
@@ -163,5 +207,36 @@ private void DeleteProducts(int first_id, int last_id)
                 id += 1;
             }
         }
+
+        private async Task<string> AssertTriggerListenerError(string functionName)
+        {
+            string errorMessage = null;
+            var tcs = new TaskCompletionSource<bool>();
+
+            void OutputHandler(object sender, DataReceivedEventArgs e)
+            {
+                if (errorMessage == null && e.Data?.Contains("Failed to start SQL trigger listener") == true)
+                {
+                    string exceptionPrefix = "Exception: System.InvalidOperationException: ";
+                    int index = e.Data.IndexOf(exceptionPrefix, StringComparison.Ordinal);
+                    Assert.NotEqual(-1, index);
```

Comment:
Why the assert here? That isn't going to really do anything since nothing is handling the exception, right? I'd suggest just doing an `if (index >= 0) {` for the following code instead.

Reply:
We are asserting on the exact log message in the respective test cases, and this on assists to those assertions by applying additional check on the log message. If I replace it with `if (index >= 0)`, a future assertion inside the test case methods would fail, which will make it slightly difficult to track the origin of the error.

Reply:
I'm not sure what you're saying. There's going to be a bunch of log messages - most of which aren't containing the error message. In those cases we should just no-op. The point here is not to fail if we get a message that doesn't contain the expected error message (since that isn't going to be the first message anyways) but rather we only care about completing the task once it sees that message. 

If it sees any other message then it should just skip it and wait for the next message.

Reply:
The other lines will be filtered out by condition `e.Data?.Contains("Failed to start SQL trigger listener") == true`. This particular assert ensures that `InvalidOperationException` is thrown.

Reply:
Hi @Charles-Gagnon, please check my last comment and see whether or not a code change will be required to resolve the conversation.

Reply:
Generally I don't find it useful to check for specific exception types unless it's a custom type (since InvalidOperationException could be thrown for a variety of reasons). But if you really think it's useful to have then it's fine to keep as is, although I would suggest appending to the comment about why we want to check for this specific exception type.